packages/vue-virtual-scroller/src/components/RecycleScroller.vue

```diff
@@ -231,19 +243,31 @@ export default {
       return view
     },
 
-    unuseView (view, fake = false) {
-      const unusedViews = this.$_unusedViews
-      const type = view.nr.type
-      let unusedPool = unusedViews.get(type)
-      if (!unusedPool) {
-        unusedPool = []
-        unusedViews.set(type, unusedPool)
+    getRecycledView (type) {
+      const recycledPool = this.getRecycledPool(type)
+      if (recycledPool && recycledPool.length) {
+        const view = recycledPool.pop()
+        view.nr.used = true
+        return view
+      } else {
+        return null
       }
-      unusedPool.push(view)
-      if (!fake) {
-        view.nr.used = false
-        view.position = -9999
-        this.$_views.delete(view.nr.key)
+    },
+
+    removeAndRecycleView (view) {
+      const type = view.nr.type
+      const recycledPool = this.getRecycledPool(type)
+      recycledPool.push(view)
+      view.nr.used = false
+      view.position = -9999
+      this.$_views.delete(view.nr.key)
+    },
+
+    removeAndRecycleAllViews () {
+      this.$_views.clear()
+      this.$_recycledPools.clear()
+      for (let i = 0, l = this.pool.length; i < l; i++) {
```

Comment:
I noticed that for is used in the original code. Why not use "Array. prototype. forEach()" to make this code more advanced?😄

Reply:
Not sure if that's better in clarity in any way.

Reply:
I think in JavaScript, using for means that "break, return, continue" will be used to jump out of the loop.

Reply:
Do you have any reference (style guide etc) for that?

Reply:
Just a suggestion.

![image](https://user-images.githubusercontent.com/44798353/188426203-97862616-41ec-4e2b-ad63-47338992d0e6.png)

[details](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach)

Reply:
Again, that's not a guidance to *use* forEach. It merely says that forEach can't be used if you want to break out of the loop.

Reply:
Just a suggestion, we can completely delete foreach in JavaScript. Because for can completely replace foreach. I mean, because for can be interrupted, using for means "I will end the loop ahead of time", and using foreach means "I will traverse the entire array". I think the reason for foreach in JavaScript is just because of the above semantics.